sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+   * Send pending invoices to CAS.
+   * @param parentProcessSummary parent process summary for logging.
+   * @param pendingInvoices pending invoices.
+   */
+
+  async sendInvoices(
+    parentProcessSummary: ProcessSummary,
+    pendingInvoices: CASInvoice[],
+  ): Promise<void> {
+    // Process each invoice in parallel.
+    await processInParallel(
+      (pendingInvoice) =>
+        this.processInvoice(pendingInvoice, parentProcessSummary),
+      pendingInvoices,
+    );
+  }
+
+  /**
+   * Process and send an invoice to CAS.
+   * @param pendingInvoice pending invoice.
+   * @param parentProcessSummary parent process summary for logging.
+   */
+  private async processInvoice(
+    pendingInvoice: CASInvoice,
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    // Log information about the pending invoice being processed.
+    parentProcessSummary.info(
+      `Processing pending invoice: ${pendingInvoice.invoiceNumber}.`,
+    );
+    const pendingInvoicePayload = this.getPendingInvoicePayload(pendingInvoice);
+    parentProcessSummary.info(
+      `Pending invoice payload invoice number: ${pendingInvoicePayload.invoiceNumber}.`,
+    );
+    let response: SendPendingInvoicesResponse;
+    try {
+      response = await this.casService.sendPendingInvoices(
+        pendingInvoicePayload,
+      );
+      response.invoiceNumber
+        ? parentProcessSummary.info(
+            `Invoice sent to CAS ${response.casReturnedMessages}.`,
+          )
+        : parentProcessSummary.warn(
+            `Invoice ${pendingInvoicePayload.invoiceNumber} send to CAS did not succeed. Reason: ${response.casReturnedMessages}.`,
+          );
+      await this.dataSource.getRepository(CASInvoice).update(
+        {
+          id: pendingInvoice.id,
+        },
+        {
+          invoiceStatus: CASInvoiceStatus.Sent,
+          invoiceStatusUpdatedOn: new Date(),
+          modifier: { id: this.systemUsersService.systemUser.id },
+        },
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        if (error.name === CAS_BAD_REQUEST) {
+          parentProcessSummary.warn("Known CAS error while sending invoice.");
+          await this.processBadRequestErrors(
+            pendingInvoice.id,
+            parentProcessSummary,
+            error.objectInfo as string[],
+            this.systemUsersService.systemUser.id,
+          );
+        }
+      } else {
+        parentProcessSummary.error(
+          "Unexpected error while sending invoice to CAS.",
+          error,
+        );
+        return null;
+      }
+    }
+  }
+
+  /**
+   * Process bad request errors from CAS API during invoice sending.
+   * @param invoiceId invoice id.
+   * @param summary summary for logging.
+   * @param error error object.
+   * @param auditUserId audit user id.
+   */
+  async processBadRequestErrors(
+    invoiceId: number,
+    summary: ProcessSummary,
+    error: string[],
+    auditUserId: number,
```

Comment:
No need to pass the `auditUserId` since it will be the system user for the entire service, right?